RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -203,14 +203,23 @@ void MahiFit::doFit(std::array<float, 3>& correctedOutput, int nbx) const {
   }
 }
 
-const float MahiFit::minimize() const {
+const float MahiFit::minimize(const double noisecorr) const {
   nnlsWork_.invcovp.setZero(nnlsWork_.tsSize, nnlsWork_.nPulseTot);
   nnlsWork_.ampVec.setZero(nnlsWork_.nPulseTot);
 
   SampleMatrix invCovMat;
   invCovMat.setConstant(nnlsWork_.tsSize, nnlsWork_.tsSize, nnlsWork_.pedVal);
   invCovMat += nnlsWork_.noiseTerms.asDiagonal();
 
+  //Add off-Diagonal components up to first order
+  if (noisecorr > 0.) {
```

Comment:
on a second thought, shouldn't this be ```!= 0``` ?
I imagine anti-correlations are technically possible

Reply:
That makes more sense. Thanks!